bears/c_languages/AStyleLintBear.py

```diff
@@ -0,0 +1,62 @@
+from coalib.bearlib.abstractions.Linter import linter
+from coalib.bearlib.spacing.SpacingHelper import SpacingHelper
+from dependency_management.requirements.DistributionRequirement import (
+    DistributionRequirement)
+
+
+@linter(executable='astyle',
+        output_format='corrected',
+        use_stdin=True)
+class AStyleLintBear:
+    """
+    Artistic Style is a source code indenter, formatter,
+    and beautifier for the C, C++, C++/CLI, Objective‑C,
+    C# and Java programming languages.
+
+    For more information, consult <http://astyle.sourceforge.net/astyle.html>.
+    """
+
+    LANGUAGES = {'C', 'C++', 'Objective-C', 'C#', 'Java'}
+    REQUIREMENTS = {DistributionRequirement(apt_get='astyle')}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    CAN_FIX = {'Formatting'}
+
+    @staticmethod
+    def create_arguments(filename, file, config_file,
+                         astyle_style: str='',
+                         use_spaces: bool=None,
+                         astyle_config: str='',
+                         indent_size: int=0):
+        """
+        :param astyle_style:
+            Defines the bracket style to use. If no bracket style is
+            specified, the default 'allman' is used.
+        :param use_spaces:
+            If None, no indentation is specified and astyle uses a default.
+            If True, astyle uses spaces or if explicitly set to False, astyle
+            uses tabs for indentation.
+        :param astyle_config:
+            Optional configuration file for astyle to read configs.
```

Comment:
nope you define the flag in `create_arguments` :)

The `config_file` parameter contains the full path to the generated config file in `generate_config`.

Anyway coala still supports setting configuration files specific to that tool, so people can migrate easier. You are right, ideally we don't want to allow those, but often those contain way more options than coala currently supports.

This changes with "aspects", where we want to streamline settings and options, and there we won't allow passing some `astyle_config` files (or config files used by other tools).